src/hooks/useDisputes.js

```diff
@@ -38,37 +40,99 @@ export default function useDisputes() {
     }
 
     return {
-      disputes: disputes.map((dispute, i) => ({
-        ...dispute,
-        ...processDisputeData(dispute),
-        ...disputesPhases[i],
-      })),
+      disputes: disputes.map((dispute, i) => {
+        const [disputeDescription] = getDisputeInfoFromMetadata(
+          dispute.metadata
+        )
+        return {
+          ...dispute,
+          description: disputeDescription,
+          ...disputesPhases[i],
+        }
+      }),
     }
   }, [disputesPhases, disputes, disputesPhasesKey, error]) // eslint-disable-line react-hooks/exhaustive-deps
 }
 
 export function useDispute(disputeId) {
+  const [disputeProcessed, setDisputeProcessed] = useState()
   const courtConfig = useCourtConfig()
   const now = useNow() // TODO: use court clock
   const { dispute, fetching } = useSingleDisputeSubscription(disputeId)
+  useEffect(() => {
+    const fetchDataFromIpfs = async () => {
+      if (dispute) {
```

Comment:
Perhaps we can do `if (!dispute) { return }` to avoid the further indentation later.